crates/revm/src/state_change.rs

```diff
@@ -46,6 +51,68 @@ pub fn post_block_balance_increments(
     balance_increments
 }
 
+/// Applies the pre-block call to the EIP-4788 beacon block root contract, using the given block,
+/// [ChainSpec], EVM.
+///
+/// If cancun is not activated or the block is the genesis block, then this is a no-op, and no
+/// state changes are made.
+#[inline]
+pub fn apply_beacon_root_contract_call<DB: Database>(
+    chain_spec: &ChainSpec,
+    block_timestamp: u64,
+    block_number: u64,
+    block_parent_beacon_block_root: Option<H256>,
+    evm: &mut EVM<State<DB>>,
+) -> Result<(), BlockExecutionError>
+where
+    <DB as Database>::Error: Debug,
+{
+    if chain_spec.is_cancun_activated_at_timestamp(block_timestamp) {
```

Comment:
you can de-indent twice here if you collapse the ifs / invert this and return early, but nbd